src/qtism/runtime/storage/binary/QtiBinaryStreamAccess.php

```diff
@@ -728,27 +738,25 @@ public function readAssessmentItemSession(
                 // in the assessment tree.
                 $varPosition = $this->readShort();
 
-                $variable = null;
-
                 try {
-                    if ($varNature === 0) {
-                        // outcome
-                        $variable = $seeker->seekComponent('outcomeDeclaration', $varPosition);
-                        $variable = OutcomeVariable::createFromDataModel($variable);
-                    } elseif ($varNature === 1) {
-                        // response
-                        $variable = $seeker->seekComponent('responseDeclaration', $varPosition);
-                        $variable = ResponseVariable::createFromDataModel($variable);
-                    } elseif ($varNature === 2) {
-                        // template
-                        $variable = $seeker->seekComponent('templateDeclaration', $varPosition);
-                        $variable = TemplateVariable::createFromDataModel($variable);
-                    }
+                    /** @var VariableDeclaration $variableDeclaration */
+                    $variableDeclaration = $seeker->seekComponent(
+                        $this->decodeVariableType($varNature),
```

Comment:
Variable type resolution should be done before the try loop. Then `decodeVariableType` method is not necessary, it's just a mapping lookup.